libraries/group_execute_provider.rb

```diff
@@ -0,0 +1,28 @@
+class Chef
+  class Provider
+    class GroupExecute < Chef::Provider::Execute
+      provides :group_execute
+
+      def_delegators :secondary_groups
+
+      private
+
+      def command
+        command = new_resource.command
+
+        if new_resource.secondary_groups
+          groups = new_resource.secondary_groups
+          if group
+            # Always do the 'primary' group last in the call chain
+            groups = [group] + groups
```

Comment:
Unless I'm misreading, it seems like the comment and statement disagree.
The comment says the primary group should be last but then concatenates an array containing just the primary group to the head of the array of groups.

Reply:
The call stack is created from the inside out, which makes the shell escaping easier. So the last group in the list is the first one to get evaluated as it works in towards the actual command. The order of the groups seems to have no bearing on the behavior, beyond the fact that the last (inner-most) call ends up setting the primary group.

Reply:
If it would make it more clear, we could make this `groups + [group]` as long as we reverse the iteration order on line 19.

Reply:
I went ahead and reversed the iteration in 702abf3.